src/watcher.ts

```diff
@@ -0,0 +1,215 @@
+import { getCustomRepository } from "typeorm";
+import {
+    UserSettingsRepository,
+    Timers,
+    Counts,
+} from "./repository/UserSettingsRepository";
+import { Client, User } from "discord.js";
+import { UserSettings } from "./entity/UserSettings";
+import * as wrapper from "./wrapper/wrapper";
+import {
+    isNotifResponse,
+    isUserResponse,
+    isCountryResponse,
+} from "./wrapper/utils";
+import { makeLogger } from "./logger";
+import { NotificationData } from "./wrapper/models/notification";
+
+const logger = makeLogger(module);
+
+export async function checkUsers(client: Client) {
+    logger.info("checking all users...");
+    const settingsRepository = getRepository();
+    const values = await settingsRepository.getAllUserSettings();
+    values?.forEach(async (user) => {
+        if (user && user.api_key) {
+            logger.info("checking " + user.user_id);
+            const notifsData = await wrapper.getNotifications(user.api_key);
+            if (isNotifResponse(notifsData?.data) && user.paused_flag != true) {
+                const discord = await client.users
+                    .fetch(user.discord_id)
+                    .catch((error) => {
```

Comment:
why what

Reply:
the `catch`. In case of an error this will go on to fail somewhere down the line in a way that *probably* won't cause problems, but why do that in the first place...

Reply:
it will literally do nothing. immediately after it checks if it exists, and if it doesnt it just stops.

Reply:
and if discord is being a pain and not working, theres not really anything to do anyways other than log it and wait for it to stop being dumb.

Reply:
but it's ugly :(